portal-api/api-tests/v1/gef/amendments/amendments.get.api-test.ts

```diff
@@ -0,0 +1,139 @@
+import { ObjectId } from 'mongodb';
+import { AMENDMENT_STATUS, AMENDMENT_TYPES, AnyObject, FacilityAmendmentWithUkefId } from '@ukef/dtfs2-common';
+import app from '../../../../src/createApp';
+import testUserCache from '../../../api-test-users';
+
+import { MAKER } from '../../../../src/v1/roles/roles';
+import { getAmendmentUrl } from './get-amendment-url';
+import createApi from '../../../api';
+import { TestUser } from '../../../types/test-user';
+
+const { as, get } = createApi(app);
+
+const getPortalFacilityAmendmentMock = jest.fn() as jest.Mock<Promise<FacilityAmendmentWithUkefId>>;
+
+jest.mock('../../../../src/v1/api', () => ({
+  ...jest.requireActual<AnyObject>('../../../../src/v1/api'),
+  getPortalFacilityAmendment: () => getPortalFacilityAmendmentMock(),
+}));
+
+const originalProcessEnv = { ...process.env };
+
+const validFacilityId = new ObjectId().toString();
+const validAmendmentId = new ObjectId().toString();
+
+const invalidId = 'invalid-id';
+
+describe('/v1/gef/facilities/:facilityId/amendments/:amendmentId', () => {
+  let testUsers: Awaited<ReturnType<typeof testUserCache.initialise>>;
+  let aMaker: TestUser;
+
+  describe('GET /v1/gef/facilities/:facilityId/amendments/:amendmentId', () => {
+    beforeEach(() => {
+      jest.resetAllMocks();
+    });
+
+    beforeAll(async () => {
+      testUsers = await testUserCache.initialise(app);
+      aMaker = testUsers().withRole(MAKER).one() as TestUser;
+    });
+
+    afterAll(() => {
+      jest.restoreAllMocks();
+      process.env = originalProcessEnv;
+    });
```

Comment:
Yes it should - copy and paste legacy over from deal-cancellation so might need to look at that there too. Probably won't change those files here to reduce file number count

Reply:
Perhaps it's intentional there - could you see what you reckon when you do a search of  `jest.restoreAllMocks();` in the codebase?

Reply:
Have changed it here!

Reply:
Ah I think if you use `reset` alongside `jest.mock` at the top of the file it gets rid of that mock & breaks the test. You cant put `jest.mock` inside a `beforeEach` though because then it happens after the function is imported

Reply:
Though in an afterAll we should be reseting all anyway so :/